Mathlib/Analysis/Complex/IntegerCompl.lean

```diff
@@ -0,0 +1,52 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.Analysis.Complex.UpperHalfPlane.Basic
+
+/-!
+# Integer Complement
+
+We define the complement of the integers in the complex plane and give some basic lemmas about it.
+We also show that the upper half plane embeds into the integer complement.
+
+-/
+
+open UpperHalfPlane
+
+/--The complement of the integers in `ℂ`. -/
+abbrev Complex.IntegerComplement := {z : ℂ | ¬ ∃ (n : ℤ), z = ↑n}
+
+namespace Complex
+
+local notation "ℂ_ℤ " => IntegerComplement
+
+noncomputable instance : UniformSpace ℂ_ℤ := instUniformSpaceSubtype
+
+instance : LocallyCompactSpace ℂ_ℤ := by
+  apply IsOpen.locallyCompactSpace
+  convert Complex.isOpen_compl_range_intCast
+  rw [IntegerComplement]
+  aesop
+
+/--The coercion from the upper half plane into the IntegerComplement. -/
+@[coe] def ucoe : ℍ → ℂ_ℤ := fun z => ⟨z, by simpa using UpperHalfPlane.ne_int z⟩
+
+instance : Coe ℍ ℂ_ℤ := ⟨ucoe⟩
```

Comment:
Keeping track of coercions is already a serious headache in the modular forms code; let's not add even more of them. Why not just `lemma UpperHalfPlane.coe_mem_integerComplement (z : ℍ) : \u z \in ℂ_ℤ`?